### PR TITLE
ASR: Support chained CompareOp

### LINKED ISSUE #2131: Missing multiadic Boolean ==
VE:

```python
from lpython import (i8, i32, i64, f32, f64,
                     dataclass
                     )
from numpy import (empty,
                   int8,
                   )

r : i8 = i8(240)

@dataclass
class Foo:
    a : i8[4] = empty(4, dtype=int8)
    dim : i32 = 4

def trinary_majority(x : Foo, y : Foo, z : Foo) -> Foo:
    foo : Foo = Foo()

    assert foo.dim == x.dim == y.dim == z.dim

    i : i32
    for i in range(foo.dim):
        pass


    # Issue 2130
    # foo.a = x.a | y.a | z.a
    return foo


t1 : Foo = Foo()
t1.a = empty(4, dtype=int8)

t2 : Foo = Foo()
t2.a = empty(4, dtype=int8)

t3 : Foo = Foo()
t3.a = empty(4, dtype=int8)

r1 : Foo = trinary_majority(t1, t2, t3)
```

runs:

```bash
(lp) ┌─(~/CLionProjects/lpython/lasr/LP-pycharm)─────────────────────────────────────────────────────────────────────────────────────────────────────────(brian@Golf37:s012)─┐
└─(11:39:11 on brian-lasr ✹ ✭)──> ~/CLionProjects/lpython/src/bin/lpython -I. issue2126.py                                                            1 ↵ ──(Sun,Jul09)─┘
semantic error: Type mismatch in comparison operator, the types must be compatible
  --> issue2126.py:18:12
   |
18 |     assert foo.dim == x.dim == y.dim == z.dim
   |            ^^^^^^^^^^^^^^^^    ^^^^^ type mismatch ('bool' and 'i32')


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
(lp) ┌─(~/CLionProjects/lpython/lasr/LP-pycharm)─────────────────────────────────────────────────────────────────────────────────────────────────────────(brian@Golf37:s012)─┐
└─(11:39:26 on brian-lasr ✹ ✭)──> PYTHONPATH='../../src/runtime/lpython' python issue2126.py
```

### PR DESCRIPTION
**Pull request by Smit-create**
Fixes #2131

### CODE REVIEW

File: src/libasr/codegen/asr_to_c_cpp.h [hunk 1]
```diff
@@ -193,6 +193,8 @@ class BaseCCPPVisitor : public ASR::BaseVisitor<Struct>
 
     void visit_TranslationUnit(const ASR::TranslationUnit_t &x) {
         global_scope = x.m_global_scope;
+        SymbolTable current_scope_copy = current_scope;
```
Comment (certik) [inline src/libasr/codegen/asr_to_c_cpp.h:196]: Shouldn't you do:
```suggestion
        SymbolTable *current_scope_copy = current_scope;
```
?

Comment (Smit-create) [reply src/libasr/codegen/asr_to_c_cpp.h]: Yes, thanks

**Review by czgdp1807 — APPROVED** [review]
LLVM already works but C needed some changes. That's surprising. Approving it anyways.

**Review by certik — APPROVED** [review]
I think that looks good.

**Pull request merged**